examples/Pin_control/Pin_control.ino

```diff
@@ -5,57 +5,176 @@
+#else
 PinI2C pins(&wrapper); // ...that the pin interface needs to communicate with the target
+#endif
 
-const uint8_t dPinIn  = 12; // any pin; connect switch against GND and +V (or use only GND and INPUT_PULLUP below)
-const uint8_t dPinOut = 13; // any pin; connect LED with resistor or just use 13 = LED_BUILTIN on Uno/Nano 
-const uint8_t aPinIn  = 14; // needs analog pin; 14 = A0 on Uno/Nano; connect potentiometer against GND and +V
+// Note: These are pins on the Target, use integers, don't use names like "A0".
+// You may have to deduce the integer value on the target with: Serial.println(A0);
+const uint8_t dPinIn  = 12; // any pin; a simple switch (SPST normally-open) connects to GND.
+const uint8_t dPinOut = 13; // any pin; connect LED with resistor or just use 13 = LED_BUILTIN on Uno/Nano
+const uint8_t aPinIn  = 14; // needs analog-in pin; 14 is A0 on Uno, connect potentiometer against GND and +V
 const uint8_t aPinOut = 6;  // needs PWM pin; 6 is PWM-capable on Uno/Nano; connect LED with resistor, or multimeter
 
-
 void setup()
 {
   Wire.begin();
   Serial.begin(115200);
+  // all chips with "native" usb require waiting till `Serial` is true
+  unsigned int begin_time=millis();
+  while(! Serial && millis()-begin_time < 1000) { delay(10); } // but at most 1 sec if not plugged in to usb
+
   // Wire.setClock(10000); // uncomment for ESP8266 targets, to be on the safe side
-  
+
+#ifndef I2CWRAPPER_LOOPBACK
   if (!wrapper.ping()) {
     halt("Target not found! Check connections and restart.");
   } else {
     Serial.println("Target found as expected. Proceeding.\n");
   }
+#else
+  Serial.print(F( "I2CWRAPPER_LOOPBACK " __FILE__ " " __DATE__ " " __TIME__ " gcc " __VERSION__ " ide "));
+  Serial.println(ARDUINO);
   
+  // A little help on the pins
+  static_assert( dPinIn < NUM_DIGITAL_PINS, "dPinIn is not a digital pin (too large)");
+  static_assert( dPinOut < NUM_DIGITAL_PINS, "dPinOut is not a digital pin (too large)");
+#ifndef ARDUINO_ARCH_SAMD
+  // sadly not usable w/samd arduino code
+  static_assert(digitalPinHasPWM(aPinOut), "aPinOut is not PWM capable on this Arduino");
+#endif
+  // doesn't cover all cases (e.g. int 13 on Uno):
+  static_assert( analogInputToDigitalPin(aPinIn - analogInputToDigitalPin(0)) >= 0, "aPinIn is not analogRead() capable an this Arduino");
+#ifdef LED_BUILTIN
+  Serial.print("LED_BUILTIN="); Serial.println(LED_BUILTIN);
+#else
+  Serial.print("LED_BUILTIN not defined on this Arduino");
+#endif
+  Serial.print("A0="); Serial.println(A0);
+  Serial.print("aPinIn "); Serial.print(aPinIn); Serial.print("=A"); Serial.println(aPinIn - analogInputToDigitalPin(0));
+
+  Serial.println("First I2C, if it hangs here, connect I2C or at least put a pull-up on SDA & SCL...");
+  if (!wrapper.ping()) {
+    Serial.println("Loopback mode, no target found AS EXPECTED. Proceeding");
+  } else {
+    Serial.println("Loopback mode, Target UNEXPECTEDLY found. Proceeding.\n");
+  }
```

Comment:
I had to duplicate the `if ping` section, because I changed the messages. I debated `#ifdef` in the `halt()` function instead.